node/src/components/binary_port.rs

```diff
@@ -452,43 +453,125 @@ where
             };
             response
         }
+        GlobalStateRequest::DictionaryItem {
+            state_identifier,
+            identifier,
+        } => {
+            let Some(state_root_hash) = resolve_state_root_hash(effect_builder, state_identifier).await else {
+                return BinaryResponse::new_empty(protocol_version);
+            };
+            let result = match identifier {
```

Comment:
Account and Contract named keys are already legacy in 2.0. I think it's fine to leave them for convenience but don't we also want a way to get a dictionary item based on the new scheme of getting named keys using `Key::NamedKey(entity_addr)` for newer contracts that are already migrated to the `AddressableEntity`?

Reply:
Or maybe it would be even more robust if we don't provide any convenience at all and only allow lookup by seed Uref and item name.

Reply:
I didn't know about the new scheme for NamedKey, it should probably be added here.
As for the convenience requests, if we still want to support the lookup in legacy accounts/contracts that logic will have to live somewhere, either the sidecar or the node, and I think it's probably better to put it in the node.

Reply:
@alsrdn I've added a variant for the new Key::NamedKey. I've also included a fix for a minor bug in TrackingCopy::query which prevented the tests from working after talking to Karan